### PR TITLE
Improve tc send 03 checks

### PR DESCRIPTION
**Pull request by kin0992**
#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/domain/checks/GetPaymentNotificationMetadataChecks.ts [hunk 1]
```diff
@@ -26,7 +28,27 @@ export const hasPAGOPAAsAttachmentNameC = flow(
   RA.exists(({ input }) => input.attachmentName === 'PAGOPA')
 );
 
+const calledDownloadEndpointC = pipe(
```
Comment (datalek) [inline src/domain/checks/GetPaymentNotificationMetadataChecks.ts:31]: WDYT about trying to unify the two `calledDownloadEndpointC`?

Comment (kin0992) [reply src/domain/checks/GetPaymentNotificationMetadataChecks.ts:31]: I totally agree with you. I'm going to take a look at it and try to unify them!

File: src/domain/checks/DownloadRecordChecks.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export const calledDownloadEndpointC = pipe(
+  R.Do,
+  R.apS('getNotificationDocumentMetadataRecordList', RA.filterMap(isGetNotificationDocumentMetadataRecord)),
+  R.apS('getPaymentNotificationMetadataRecordList', RA.filterMap(isGetPaymentNotificationMetadataRecord)),
+  R.apS('downloadRecordList', RA.filterMap(isDownloadRecord)),
+  R.map(({ getNotificationDocumentMetadataRecordList, getPaymentNotificationMetadataRecordList, downloadRecordList }) =>
+    pipe(
+      RA.concatW(getNotificationDocumentMetadataRecordList)(getPaymentNotificationMetadataRecordList),
+      RA.filterMap(({ output }) => (output.statusCode === 200 ? O.fromNullable(output.returned.url) : O.none)),
+      RA.every((url) =>
```
Comment (kin0992) [inline src/domain/checks/DownloadRecordChecks.ts:18]: I also changed this method from `RA.exists` to `RA.every`. WDYT it is better @datalek?

File: src/domain/checks/GetPaymentNotificationMetadataChecks.ts (hunk 1 above)
Comment (kin0992) [reply src/domain/checks/GetPaymentNotificationMetadataChecks.ts]: Pushed some refactor

**Review by datalek — APPROVED** [review]
👍

**Pull request merged**